garden-backend-service/tests/api/routes/test_gardens.py

```diff
@@ -815,3 +816,73 @@ async def test_search_gardens_offset_paginates_results(
     assert result["total"] == 20
     assert result["count"] == 5
     assert result["offset"] == 15
+
+
+@pytest.mark.asyncio
+@pytest.mark.integration
+async def test_search_gardens_sort_by_title(
+    client,
+    mock_db_session,
+    override_authenticated_dependency,
+    mock_garden_create_request_no_entrypoints_json,
+):
+    sorted_titles = ["A", "B", "C", "D", "E", "F"]
+
+    random_titles = list(sorted_titles)
+    random.shuffle(random_titles)
```

Comment:
I'm not too worried about false positives here (there's only a 1 out of 6 factorial chance the list stays in place, and the asc vs desc logic mitigates that being a problem) but generally I don't think we should inject non-determinism into this kind of test

Reply:
Good catch! I don't want to think too hard about this, but a bit of a crude solution is to do:
```
while random_titles == sorted_titles:
    random.shuffle(random_titles)
```
I suppose there is a small chance of an infinite loop, but I think the average number of iterations will be <2

Reply:
Ha, I like it